src/corrupt.cc

```diff
@@ -823,7 +823,8 @@ static bool test_depend_corrupt(s16b corrupt_idx, bool can_gain)
 	}
 
 	/* Go through all dependencies */
-	for (i=0; c_ptr->depends[i] >= 0; i++)
+	for (i=0; i < sizeof(c_ptr->depends)/sizeof(c_ptr->depends[0])
```

Comment:
There's actually a std::size() which does the right thing for arrays.

That said, it's pretty weird that this would fix anything. AFAICT all the depends/opposes arrays in the 
`corruptions` array seem to be properly terminated with a -1 (and none have more than 4 elements). 
Maybe I'm missing something, though. Could you point to where in the array you see the "bad" entry?